src/ESM.sol

```diff
@@ -21,31 +21,37 @@ pragma solidity >=0.6.12;
 
 interface GemLike {
     function balanceOf(address) external view returns (uint256);
+    function burn(uint256) external;
     function transfer(address, uint256) external returns (bool);
     function transferFrom(address, address, uint256) external returns (bool);
 }
 
 interface EndLike {
+    function live() external view returns (uint256);
+    function vat()  external view returns (address);
     function cage() external;
 }
 
+interface VatLike {
+    function deny(address) external;
+}
+
 contract ESM {
-    GemLike public immutable gem; // collateral
-    EndLike public immutable end; // cage module
-    address public immutable pit; // burner
-    uint256 public immutable min; // threshold
-    uint256 public fired;
+    GemLike public immutable gem;   // collateral
+    EndLike public immutable end;   // cage module
+    address public immutable proxy; // Pause proxy
+    uint256 public immutable min;   // threshold
```

Comment:
Consider updating comment here to `minimum MKR deposit threshold [wad]`

Reply:
or `gem` for `MKR`

Reply:
Done!